src/solvers/refinement/string_refinement.cpp

```diff
@@ -257,64 +399,65 @@ bool string_refinementt::is_char_array(const typet &type) const
 /// \param rhs: right and side of the equality
 /// \return true if the assignemnt needs to be handled by the parent class
 ///         via `set_to`
-bool string_refinementt::add_axioms_for_string_assigns(
-  const exprt &lhs, const exprt &rhs)
+bool add_axioms_for_string_assigns(
+  replace_mapt& symbol_resolve,
+  std::map<exprt, std::list<exprt>> &reverse_symbol_resolve,
+  string_constraint_generatort& generator,
+  messaget::mstreamt &stream,
+  const namespacet &ns,
+  const exprt &lhs,
+  const exprt &rhs)
 {
-  if(is_char_array(rhs.type()))
+  if(is_char_array(ns, rhs.type()))
   {
-    set_char_array_equality(lhs, rhs);
```

Comment:
I think it's important that this function call stays here since it's an integral part of the process this function performs.

Reply:
Original function call location has been restored.